feat(community-testimonial): add testimonial component

agorovyi opened the pull request:
<!--
  ### IMPORTANT SECURITY NOTE ###

  When opening pull requests, be sure NOT to include any private or personal
  information such as secrets, passwords, or any source code that involves
  data retrieval.
-->

## Related issues

See #90 

## Package changes

```
yarn run v1.9.2
$ yarn test:e2e && node scripts/prePr.js --conventional-commits --yes
$ node scripts/e2e.js
Started chromedriver
Starting ChromeDriver 2.38.552518 (183d19265345f54ce39cbb94cf81ba5f15905011) on port 9515
Only local connections are allowed.

[Components Spec] Test Suite
================================

Running:  @tds/community-testimonial
 ✔ Element <#testimonial> was visible after 26 milliseconds.
 ✔ Passed [ok]: 11 aXe Tests Passed
 ✔ #rsg-root passes accessibility scan
WARNING: baseline photo does NOT exist.
Creating baseline photo from result: /Users/x207090/td/tds-community/e2e/output/components/Testimonial/baseline/chrome_headless.png
 ✔ Screenshots Matched for chrome_headless.png with a tolerance of 0%, actual was 0%.

OK. 4 assertions passed. (2.616s)

Stopped chromedriver
lerna info versioning independent
lerna info ignore [ '*.md', '*.spec.jsx', '*.snap' ]
lerna info Checking for updated packages...
lerna info Comparing with @tds/community-sample-pilter@1.0.0.
lerna info Checking for prereleased packages...

Changes:
 - @tds/community-testimonial: 1.0.0 => 1.1.0

lerna info auto-confirmed

🤔 If this is not what you expected, ensure that your commit messages follow the Conventional Commits specification (https://conventionalcommits.org) and try again.

🚀 Please paste the entire output of this task into the body of your Pull Request so that a maintainer can verify before merging/publishing.
✨  Done in 4.98s.
```

## Checklist before submitting pull request

* Commits follow our [Developer Guide](https://github.com/telus/tds-community/blob/chore/template-hints/.github/CONTRIBUTING.md)
* [ ] New code is unit tested
* [ ] For code changes, run `yarn prepr` locally
  * make sure visual and accessibility tests pass
  * paste the lerna version output

jesdavpet commented on packages/Testimonial/Testimonial.jsx:
```diff
@@ -0,0 +1,46 @@
+    <Card {...safeRest(rest)}>
+      <Box between={3}>
+        <Paragraph>&quot;{testimonial}&quot;</Paragraph>
+        <Box inline={3} dangerouslyAddClassName={styles['testimonial-author']}>
```
It looks like this line is causing a PropTypes validation error: `<Box inline={3} ...>`

<img width="737" alt="screen shot 2018-09-28 at 3 05 31 pm" src="https://user-images.githubusercontent.com/5464587/46228599-8a3e8600-c330-11e8-8bce-34b41aed2a14.png">

jesdavpet commented on packages/Testimonial/Testimonial.jsx:
```diff
@@ -0,0 +1,46 @@
+  return (
+    <Card {...safeRest(rest)}>
+      <Box between={3}>
+        <Paragraph>&quot;{testimonial}&quot;</Paragraph>
```
The use of `&quot;` as a hard-coded value actually isn't `i18n` friendly, different languages use different characters as quotation marks.

**Recommendation:**
By simplifying: ~`<Paragraph>&quot;{testimonial}&quot;</Paragraph>`~
to: `<Paragraph>{testimonial}</Paragraph>`
we can allow the consumer of the component to provide whatever quotation marks are appropriate for their `i18n` use case in the value passed in the  `testimonial` property.

Furthermore, it would be more typographically correct to use `&ldquo;` and `&rdquo;` for English, and (_I think_) `&laquo;` and `&raquo;` in French.

agorovyi replied:
fixed

agorovyi commented on packages/Testimonial/Testimonial.jsx:
```diff
@@ -0,0 +1,46 @@
+
+Testimonial.propTypes = {
+  /** Testimonial main text */
+  testimonial: PropTypes.string.isRequired,
```
Not 100% sure about allowing `PropTypes.element` or `PropTypes.node`. Wouldn't that allow too much of a flexibility to it like passing an entire React Component in there for example?

jesdavpet replied:
I guess it might be good to ask design, if the `testimonial` in this layout:
* will _**never**_ have multiple paragraphs?
* will _**never**_ have links?
* will _**never**_ have superscript (legal text annotation)?
* will _**never**_ have formatted text (bold, italic, etc.)?

If the component would *_ever_* need to include *_any_* of those elements, then we should allow `PropTypes.node` as well. There's a risk that if we make it too strict / rigid, then outcome teams won't adopt the component.

I support starting out with `PropTypes.string` for now, and then opening it up to also accept `PropTypes.node` etc. later, since that would be a non-breaking change, so it could just be a patch release.

But, in any case, this is just a comment / open question. Not blocking the PR for this at all. 😃

agorovyi commented:
Thanks @jesdavpet for the comments. All valid points that I've totally missed. I refactored the component to address comments and while doing it found an opportunity to simplify the code a bit. Removed custom styles as they are not needed at all. It's consisted from 100% pure core components now.

jesdavpet commented:
Awesome, this is nice work! :+1: Thanks @agorovyi !!!

joperron commented:
Hey guys, sorry to bother again, when can we expect this to be merged? I have people breathing down my neck :)

varunj90 commented:
hey @joperron we are doing this shortly, in a few hours.
